DeanerySystem/DeanerySystem.DataAccess/Configurations/DepartmentConfiguration.cs

```diff
@@ -0,0 +1,39 @@
+﻿using DeanerySystem.DataAccess.Entities;
+using DeanerySystem.DataAccess.Utilities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace DeanerySystem.DataAccess.Configurations
+{
+    class DepartmentConfiguration : DbEntityConfiguration<Department>
+    {
+        public override void Configure(EntityTypeBuilder<Department> entity)
+        {
+			entity.ToTable("Departments");
+			entity.HasKey(d => d.Id);
+			entity.Property(d => d.Id).ValueGeneratedOnAdd();
+
+            entity.HasOne(d => d.Stream)
+                .WithMany(s => s.Departments)
+                .IsRequired(true)
+                .HasForeignKey(d => d.StreamId)
+                .OnDelete(DeleteBehavior.Restrict);
```

Comment:
What does it mean? `Restrict` delete behaviour?

Reply:
You cannot delete stream as long as there is departments in this stream